File: solutions/go/sum-of-multiples/2/sum_of_multiples.go
package summultiples

func SumMultiples(limit int, divisors ...int) int {
	multiples := make([]int, limit)
	for _, div := range divisors {
		if div == 0 {
			continue
		}
		for i := 0; i < limit; i += div {
			multiples[i] = i
		}
	}

	sum := 0
	for _, val := range multiples {
		sum += val
	}
	return sum
}

File: solutions/csharp/sum-of-multiples/2/SumOfMultiples.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class SumOfMultiples
{
    public static int Sum(IEnumerable<int> multiples, int max) =>
        Enumerable.Range(0, max)
                  .Where(val => multiples.Any(mul => mul != 0 && val % mul == 0))
                  .Sum();
}
